spot_wrapper/calibration/README.md

```diff
@@ -0,0 +1,242 @@
+to a shared frame.
+
+
+# Using the CLI Tool To Calibrate On an Existing Dataset
+To use the CLI Tool, please ensure that you have one parent folder, 
+where each camera has a folder under the parent (Numbered from 0 to N). Synchronized photos 
+for each camera should appear in their respective directories, where matching photos have ids, ascending
+upwards, starting from 0. The file structure should appear something like the following:
+```
+existing_dataset/
+├── 0/
+│   ├── 0.png # taken at viewpoint 0 
+│   ├── 1.png
+│   └── 2.png
+├── 1/
+│   ├── 0.png # taken at viewpoint 0
+│   ├── 1.png
+│   └── 2.png
+├── 2/
+│   ├── 0.png # taken at viewpoint 0
+│   ├── 1.png
+│   └── 2.png
+```
+
+To see all possible arguments for calibration, please run  ```python3 calibrate_multistereo_cameras_with_charuco_cli.py -h```.
+    Many parameters such as board proportions and Agmruco dictionary are customizable.
+
+If you'd like to register camera 1 to camera 0, and camera 2 to camera 0, you could do the following:
+```
+python3 calibrate_multistereo_cameras_with_charuco_cli.py --data_path ~/existing_dataset/ \
+--result_path ~/existing_dataset/eye_in_hand_calib.yaml --photo_utilization_ratio 1 --stereo_pairs "[(1,0), (2, 0)]" \
+--tag default --unsafe_tag_save
+```
+
+# Understanding the Output Calibration Config File from the CLI
+A calibration produced with ```multistereo_calibration_charuco``` from ```calibration_util```
+can be saved as a ```.yaml```file with ```save_calibration_parameters``` from ```calibration_util```. Each calibration run
+that creates or modifies a config file is tagged with a unique name, to allow for tracking of several experiments, which is the main title.
+Under the main title, there are the fields relevant to the calibration
+Under ```intrinsic```, the intrinsic (camera matrix, distortion coefficents, and image height/width)
+for each camera are recorded as a flattened representation
+under the camera's index number as it appears in the list of images 
+returned by ```capture_images```. For example, if ```capture_images``` produces a list that is 
+```[image_by_primary_camera, image_by_reference_camera]```, then the intrinsic for the ```primary_camera```
+would be stored under ```0```, and the intrinsic for ```reference_camera``` would be stored under ```1```.
+
+Under ```extrinsic```, the first sublevel, again a camera index, corresponds to the camera index of which camera
+is the origin for the ```extrinsic``` transform between two cameras as the first
+field in a requested stereo pair. The second sublevel corresponds to the index of which camera
+the ```extrinsic``` maps to from the origin camera. Going with the example mentioned above, an extrinsic
+would be stored as 
+
+```
+extrinsic:
+    0: # primary camera index
+        1: # reference camera index
+            R: flattened_3x3_rotation_matrix_from_primary_to_reference_camera_now_9x1
+            T: 3x1_translation_matrix_from_primary_to_reference_camera
+```
```

Comment:
I'm thinking for readability it might make more sense to put the example first here, then the explanations afterwards. It was a little hard for me to follow the beginning of this without having the example in mind.

Reply:
Yeah this tracks I'm thinking about putting in an entire sample too not just extrinsic